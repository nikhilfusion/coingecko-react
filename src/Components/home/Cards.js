import React from 'react';
import {
  Grid,
  makeStyles,
  Card,
  CardMedia,
  Typography,
  CardContent,
} from '@material-ui/core';
import PropTypes from 'prop-types';
import { capitalizeName } from '../../utils/common';
// import { capitalizeName } from '../../utils/common';
// import Activity from '../common/Activity';

const useStyles = makeStyles((theme) => ({
  card: {
    display: 'flex',
    flexDirection: 'column',
    cursor: 'pointer'
  },
  cardMedia: {
    height: '250px',
    width: '250px',
    margin: 'auto'
  },
  cardContent: {
    flexGrow: 1,
  },
  contentMain: {
    alignItems: 'center'
  },
  contentHeading: {
    padding: '16px 0',
    fontWeight: 'bold'
  },
  contentValue: {
    padding: '16px 0'
  },
  green: {
    background: '#0e7d0e'
  },
  red: {
    background: '#ef3919ed'
  },
  blue: {
    background: '#3021e8'
  },
  tagContainer: {
    fontWeight: '700',
    paddingTop: '16px'
  }
}));

export default function Cards({ cardData, onCardClick }) {
  const classes = useStyles();
  const { image, name, id, symbol, current_price, high_24h, low_24h } = cardData;
  return (
    <Card className={classes.card} onClick={() => onCardClick(id)}>
      <CardMedia
        className={classes.cardMedia}
        image={image}
        title={name}
      />
      <CardContent className={classes.cardContent}>
        <Typography gutterBottom variant="h5" component="h2" align="center">
          {capitalizeName(name)}
        </Typography>
        <Grid container className={classes.contentMain}>
          <Grid item xs={8} className={classes.contentHeading}>Symbol</Grid>
          <Grid item xs={4} className={classes.contentValue}>{symbol}</Grid>
          <Grid item xs={8} className={classes.contentHeading}>Current Price</Grid>
          <Grid item xs={4} className={classes.contentValue}>{current_price}</Grid>
          <Grid item xs={8} className={classes.contentHeading}>Lowest 24hr Price</Grid>
          <Grid item xs={4} className={classes.contentValue}>{low_24h}</Grid>
          <Grid item xs={8} className={classes.contentHeading}>Highest 24hr Price</Grid>
          <Grid item xs={4} className={classes.contentValue}>{high_24h}</Grid>
        </Grid>
      </CardContent>
    </Card>
  )
}

Cards.propTypes = {
  cardData: PropTypes.object,
  onCardClick: PropTypes.func
}