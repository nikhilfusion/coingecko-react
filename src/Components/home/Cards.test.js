import React from 'react';
import Cards from './Cards';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

const mockData = {
  image: 'https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579',
  name: 'bitcoin',
  id: 'bitcoin',
  symbol: 'btc',
  current_price: '29173',
  high_24h: '30759',
  low_24h: '29005',
}

test("render Coin Data while rendering Card component", () => {
  const { getByText } = render(<Cards cardData={mockData} />);
  expect(getByText(mockData.symbol)).toBeDefined();
  expect(getByText(mockData.current_price)).toBeDefined();
  expect(getByText(mockData.high_24h)).toBeDefined();
  expect(getByText(mockData.low_24h)).toBeDefined();
})

test('it renders correctly', () => {
  const tree = renderer.create(<Cards cardData={mockData} />).toJSON();
  expect(tree).toMatchSnapshot()
})