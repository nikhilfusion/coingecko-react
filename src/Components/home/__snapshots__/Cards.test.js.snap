// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-7 MuiPaper-elevation1 MuiPaper-rounded"
  onClick={[Function]}
>
  <div
    className="MuiCardMedia-root makeStyles-cardMedia-8"
    style={
      Object {
        "backgroundImage": "url(\\"https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579\\")",
      }
    }
    title="bitcoin"
  />
  <div
    className="MuiCardContent-root makeStyles-cardContent-9"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom MuiTypography-alignCenter"
    >
      Bitcoin
    </h2>
    <div
      className="MuiGrid-root makeStyles-contentMain-10 MuiGrid-container"
    >
      <div
        className="MuiGrid-root makeStyles-contentHeading-11 MuiGrid-item MuiGrid-grid-xs-8"
      >
        Symbol
      </div>
      <div
        className="MuiGrid-root makeStyles-contentValue-12 MuiGrid-item MuiGrid-grid-xs-4"
      >
        btc
      </div>
      <div
        className="MuiGrid-root makeStyles-contentHeading-11 MuiGrid-item MuiGrid-grid-xs-8"
      >
        Current Price
      </div>
      <div
        className="MuiGrid-root makeStyles-contentValue-12 MuiGrid-item MuiGrid-grid-xs-4"
      >
        29173
      </div>
      <div
        className="MuiGrid-root makeStyles-contentHeading-11 MuiGrid-item MuiGrid-grid-xs-8"
      >
        Lowest 24hr Price
      </div>
      <div
        className="MuiGrid-root makeStyles-contentValue-12 MuiGrid-item MuiGrid-grid-xs-4"
      >
        29005
      </div>
      <div
        className="MuiGrid-root makeStyles-contentHeading-11 MuiGrid-item MuiGrid-grid-xs-8"
      >
        Highest 24hr Price
      </div>
      <div
        className="MuiGrid-root makeStyles-contentValue-12 MuiGrid-item MuiGrid-grid-xs-4"
      >
        30759
      </div>
    </div>
  </div>
</div>
`;
